pkg/controller/secret/secret_controller.go

```diff
@@ -529,11 +555,35 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 			log.Info("INFO: Cluster is not ready; skipping Pager Duty configuration")
 		}
 	}
+
 	if snitchSecretExists {
 		log.Info("INFO: Dead Man's Snitch secret exists")
 		watchdogURL = readSecretKey(r, &request, secretNameDMS, secretKeyDMS)
 	}
 
+	if namespaceCMExists {
+		log.Info("INFO: Managed namespace configMap exists")
+		var managedNamespacesConfig alertmanager.ManagedNamespacesConfig
+		rawManagedNamespaces := readCMKey(r, &request, cmNameManagedNamespaces, cmKeyManagedNamespaces)
+
+		err := yaml.Unmarshal([]byte(rawManagedNamespaces), &managedNamespacesConfig)
+		if (err != nil) || (len(managedNamespacesConfig.Resources.Namespace) == 0) {
```

Comment:
`yaml.Unmarshal` doesn't throw an error if the `managedNamespacesConfig` struct doesn't match, checking if the namespace list is empty seems to be the easiest way to determine if something is off with the yaml

Reply:
Since this should be noted for the future, I added a comment clarifying this.